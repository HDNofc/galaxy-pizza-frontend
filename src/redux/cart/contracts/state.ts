import { IPizzaCrustTypes, IPizzaDoughTypes } from 'redux/store/types';

export interface PizzaCartItem {
  _dbId: string;
  cartId: string;
  name: string;
  description: string;
  price: number;
  dough: IPizzaDoughTypes;
  crust: IPizzaCrustTypes;
  size: number;
  imageUrl?: {
    crop: {
      jpg: string;
      webp: string;
    };
    full: {
      jpg: string;
      webp: string;
    };
  };
  amount: number;
}

export interface PizzaCartState {
  items: PizzaCartItem[];
  totalPrice: number;
  totalCount: number;
}
