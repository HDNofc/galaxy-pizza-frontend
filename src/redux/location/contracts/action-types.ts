import { Action } from 'redux';

export enum LocationActionsType {
  SET_LOCATION = 'location/SET_LOCATION',
}

export interface SetLocationActionInterface extends Action<LocationActionsType> {
  type: LocationActionsType.SET_LOCATION;
  payload: string;
}
