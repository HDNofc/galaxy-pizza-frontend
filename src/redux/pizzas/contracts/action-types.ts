import { Action } from 'redux';
import { IPizza } from 'redux/store/types';

export enum PizzasActionsType {
  SET_PIZZAS = 'pizzas/SET_PIZZAS',
  CLEAR_PIZZAS = 'pizzas/CLEAR_PIZZAS',
  SET_LOADED = 'pizzas/SET_LOADED',
}

export interface SetPizzasActionInterface extends Action<PizzasActionsType> {
  type: PizzasActionsType.SET_PIZZAS;
  payload: IPizza[];
}

export interface ClearPizzasActionInterface extends Action<PizzasActionsType> {
  type: PizzasActionsType.CLEAR_PIZZAS;
}

export interface SetLoadedActionInterface extends Action<PizzasActionsType> {
  type: PizzasActionsType.SET_LOADED;
  payload: boolean;
}
