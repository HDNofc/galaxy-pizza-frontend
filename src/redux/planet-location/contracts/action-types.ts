import { Action } from 'redux';

export enum PlanetLocationActionsType {
  SET_PLANET_LOCATION = 'location/SET_PLANET_LOCATION',
}

export interface SetLocationActionInterface extends Action<PlanetLocationActionsType> {
  type: PlanetLocationActionsType.SET_PLANET_LOCATION;
  payload: string;
}
