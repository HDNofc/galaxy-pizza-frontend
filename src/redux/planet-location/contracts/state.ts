export interface IPlanetLocation {
  planetLocation: string;
}
