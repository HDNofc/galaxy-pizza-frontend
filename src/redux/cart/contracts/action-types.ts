import { Action } from 'redux';
import { PizzaCartItem } from './state';

export enum CartActionsType {
  ADD_PIZZA_TO_CART = 'cart/ADD_PIZZA_TO_CART',
  PLUS_CART_ITEM = 'cart/PLUS_CART_ITEM',
  MINUS_CART_ITEM = 'cart/MINUS_CART_ITEM',
  REMOVE_CART_ITEM = 'cart/REMOVE_CART_ITEM',
  CLEAR_CART = 'cart/CLEAR_CART',
}

export interface AddPizzaToCartActionInterface extends Action<CartActionsType> {
  type: CartActionsType.ADD_PIZZA_TO_CART;
  payload: PizzaCartItem;
}

export interface PlusCartItemActionInterface extends Action<CartActionsType> {
  type: CartActionsType.PLUS_CART_ITEM;
  payload: number;
}

export interface MinusCartItemActionInterface extends Action<CartActionsType> {
  type: CartActionsType.MINUS_CART_ITEM;
  payload: number;
}
export interface RemoveCartItemActionInterface extends Action<CartActionsType> {
  type: CartActionsType.REMOVE_CART_ITEM;
  payload: number;
}

export interface ClearCartActionInterface extends Action<CartActionsType> {
  type: CartActionsType.CLEAR_CART;
}
