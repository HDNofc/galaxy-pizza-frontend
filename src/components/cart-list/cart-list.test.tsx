import userEvent from '@testing-library/user-event';
import { PizzaCartItem } from 'redux/cart/contracts/state';
import renderWithThemeProvider from 'test-utils/render-with-theme-provider';
import CartList from './cart-list';
import items from './__mocks__/';

const defaultProps = {
  items: items as PizzaCartItem[],
  onDeleteItemClick: () => {},
  onCounterPlusClick: () => {},
  onCounterMinusClick: () => {},
};

describe('<CartList>', () => {
  it('отображается корректно с верным количеством элементов', () => {
    const { container } = renderWithThemeProvider(<CartList {...defaultProps} />);

    expect(container.firstChild).toMatchSnapshot();
    expect(container.querySelectorAll('li')).toHaveLength(3);
  });

  it('коллбеки вызываются корректно', () => {
    const onDeleteItemClick = jest.fn();
    const onCounterPlusClick = jest.fn();
    const onCounterMinusClick = jest.fn();

    const { container } = renderWithThemeProvider(
      <CartList
        {...defaultProps}
        onDeleteItemClick={onDeleteItemClick}
        onCounterPlusClick={onCounterPlusClick}
        onCounterMinusClick={onCounterMinusClick}
      />
    );

    const firstPizza = container.querySelectorAll('li')[0];
    const deleteFirstPizzaButton = firstPizza.querySelectorAll('button')[0];
    const decrementFirstPizzaButton = firstPizza.querySelectorAll('button')[1];
    const incrementFirstPizzaButton = firstPizza.querySelectorAll('button')[2];

    userEvent.click(deleteFirstPizzaButton);
    expect(onDeleteItemClick).toHaveBeenCalled();

    userEvent.click(decrementFirstPizzaButton);
    expect(onCounterMinusClick).toHaveBeenCalled();

    userEvent.click(incrementFirstPizzaButton);
    expect(onCounterPlusClick).toHaveBeenCalled();
  });
});
