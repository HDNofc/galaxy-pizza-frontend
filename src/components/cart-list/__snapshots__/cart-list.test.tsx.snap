// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartList> отображается корректно с верным количеством элементов 1`] = `
<ul
  class="CartList-sc-hb6owy-0 jBTWLT"
>
  <li
    class="CartListItem-sc-hb6owy-1 bUxpHq"
  >
    <div
      class="ItemBody-sc-hb6owy-2 bCXgRP"
    >
      <div
        class="ImagePlaceholder-sc-hb6owy-4 dmvjGy"
      >
        <div
          class="ImagePlaceholder-sc-1lq6aod-0"
        >
          <div
            class="Placeholder-sc-1lq6aod-1 eizUTG"
          >
            <svg
              class="pizza-placeholder-image"
            >
              pizza-placeholder.svg
            </svg>
          </div>
          <div
            class="Content-sc-1lq6aod-2 iLQvsD"
          >
            <picture
              class="Picture-sc-hb6owy-3 lfkRa-D"
            >
              <source
                srcset=""
                type="image/webp"
              />
              <img
                alt=""
                class="CartListImage-sc-hb6owy-5 iHzSkv"
                src=""
              />
            </picture>
          </div>
        </div>
      </div>
      <div
        class="ItemInfo-sc-hb6owy-6 gdDXea"
      >
        <span
          class="ItemName-sc-hb6owy-7 fZhgMN"
        >
          Мясное барбекю
        </span>
        <p
          class="ItemShortDescription-sc-hb6owy-8 eYepFn"
        >
          Традиционное тесто, 35 см, колбасный борт
        </p>
        <p
          class="ItemFullDescription-sc-hb6owy-9 bllVyv"
        >
          Мясная пицца с пикантной пепперони, альпийскими колбасками, Моцареллой, луком и соусом Барбекю
        </p>
      </div>
      <button
        class="Button-sc-1fut0te-0 jOZQxY DeleteButton-sc-hb6owy-10 elyAfU"
        type="button"
      >
        <span
          class="Text-sc-1fut0te-1"
        >
          Удалить пиццу из корзины
        </span>
        <svg
          class="cart-list-item-cross"
        >
          cross.svg
        </svg>
      </button>
    </div>
    <div
      class="ItemSummaryPrice-sc-hb6owy-11 czNIxm"
    >
      <div
        class="Counter-sc-9bdt3x-0 fPNYnq"
      >
        <button
          class="Button-sc-1fut0te-0 DIwKh"
          type="button"
        >
          <span
            class="Text-sc-1fut0te-1"
          >
            Отнять пиццу
          </span>
          <svg>
            minus.svg
          </svg>
        </button>
        <span
          class="Value-sc-9bdt3x-1 jOvzOn"
        >
          2
        </span>
        <button
          class="Button-sc-1fut0te-0 DIwKh"
          type="button"
        >
          <span
            class="Text-sc-1fut0te-1"
          >
            Добавить пиццу
          </span>
          <svg>
            plus.svg
          </svg>
        </button>
      </div>
      <span
        class="ItemPrice-sc-hb6owy-12 iQvgly"
      >
        1 168 ₽
      </span>
    </div>
  </li>
  <li
    class="CartListItem-sc-hb6owy-1 bUxpHq"
  >
    <div
      class="ItemBody-sc-hb6owy-2 bCXgRP"
    >
      <div
        class="ImagePlaceholder-sc-hb6owy-4 dmvjGy"
      >
        <div
          class="ImagePlaceholder-sc-1lq6aod-0"
        >
          <div
            class="Placeholder-sc-1lq6aod-1 eizUTG"
          >
            <svg
              class="pizza-placeholder-image"
            >
              pizza-placeholder.svg
            </svg>
          </div>
          <div
            class="Content-sc-1lq6aod-2 iLQvsD"
          >
            <picture
              class="Picture-sc-hb6owy-3 lfkRa-D"
            >
              <source
                srcset=""
                type="image/webp"
              />
              <img
                alt=""
                class="CartListImage-sc-hb6owy-5 iHzSkv"
                src=""
              />
            </picture>
          </div>
        </div>
      </div>
      <div
        class="ItemInfo-sc-hb6owy-6 gdDXea"
      >
        <span
          class="ItemName-sc-hb6owy-7 fZhgMN"
        >
          Маленькая италия
        </span>
        <p
          class="ItemShortDescription-sc-hb6owy-8 eYepFn"
        >
          Тонкое тесто, 23 см
        </p>
        <p
          class="ItemFullDescription-sc-hb6owy-9 bllVyv"
        >
          Сыр моцарелла, пикантная пепперони, ароматная свинина, орегано, шампиньоны, черные оливки с фирменным томатным соусом
        </p>
      </div>
      <button
        class="Button-sc-1fut0te-0 jOZQxY DeleteButton-sc-hb6owy-10 elyAfU"
        type="button"
      >
        <span
          class="Text-sc-1fut0te-1"
        >
          Удалить пиццу из корзины
        </span>
        <svg
          class="cart-list-item-cross"
        >
          cross.svg
        </svg>
      </button>
    </div>
    <div
      class="ItemSummaryPrice-sc-hb6owy-11 czNIxm"
    >
      <div
        class="Counter-sc-9bdt3x-0 fPNYnq"
      >
        <button
          class="Button-sc-1fut0te-0 DIwKh"
          type="button"
        >
          <span
            class="Text-sc-1fut0te-1"
          >
            Отнять пиццу
          </span>
          <svg>
            minus.svg
          </svg>
        </button>
        <span
          class="Value-sc-9bdt3x-1 jOvzOn"
        >
          1
        </span>
        <button
          class="Button-sc-1fut0te-0 DIwKh"
          type="button"
        >
          <span
            class="Text-sc-1fut0te-1"
          >
            Добавить пиццу
          </span>
          <svg>
            plus.svg
          </svg>
        </button>
      </div>
      <span
        class="ItemPrice-sc-hb6owy-12 iQvgly"
      >
        639 ₽
      </span>
    </div>
  </li>
  <li
    class="CartListItem-sc-hb6owy-1 bUxpHq"
  >
    <div
      class="ItemBody-sc-hb6owy-2 bCXgRP"
    >
      <div
        class="ImagePlaceholder-sc-hb6owy-4 dmvjGy"
      >
        <div
          class="ImagePlaceholder-sc-1lq6aod-0"
        >
          <div
            class="Placeholder-sc-1lq6aod-1 eizUTG"
          >
            <svg
              class="pizza-placeholder-image"
            >
              pizza-placeholder.svg
            </svg>
          </div>
          <div
            class="Content-sc-1lq6aod-2 iLQvsD"
          >
            <picture
              class="Picture-sc-hb6owy-3 lfkRa-D"
            >
              <source
                srcset=""
                type="image/webp"
              />
              <img
                alt=""
                class="CartListImage-sc-hb6owy-5 iHzSkv"
                src=""
              />
            </picture>
          </div>
        </div>
      </div>
      <div
        class="ItemInfo-sc-hb6owy-6 gdDXea"
      >
        <span
          class="ItemName-sc-hb6owy-7 fZhgMN"
        >
          Веганская с овощами
        </span>
        <p
          class="ItemShortDescription-sc-hb6owy-8 eYepFn"
        >
          Традиционное тесто, 30 см
        </p>
        <p
          class="ItemFullDescription-sc-hb6owy-9 bllVyv"
        >
          Фирменный томатный соус, веганская Моцарелла, сладкий зелёный перец, шампиньоны, маслины, лук и сочные томаты
        </p>
      </div>
      <button
        class="Button-sc-1fut0te-0 jOZQxY DeleteButton-sc-hb6owy-10 elyAfU"
        type="button"
      >
        <span
          class="Text-sc-1fut0te-1"
        >
          Удалить пиццу из корзины
        </span>
        <svg
          class="cart-list-item-cross"
        >
          cross.svg
        </svg>
      </button>
    </div>
    <div
      class="ItemSummaryPrice-sc-hb6owy-11 czNIxm"
    >
      <div
        class="Counter-sc-9bdt3x-0 fPNYnq"
      >
        <button
          class="Button-sc-1fut0te-0 DIwKh"
          type="button"
        >
          <span
            class="Text-sc-1fut0te-1"
          >
            Отнять пиццу
          </span>
          <svg>
            minus.svg
          </svg>
        </button>
        <span
          class="Value-sc-9bdt3x-1 jOvzOn"
        >
          1
        </span>
        <button
          class="Button-sc-1fut0te-0 DIwKh"
          type="button"
        >
          <span
            class="Text-sc-1fut0te-1"
          >
            Добавить пиццу
          </span>
          <svg>
            plus.svg
          </svg>
        </button>
      </div>
      <span
        class="ItemPrice-sc-hb6owy-12 iQvgly"
      >
        799 ₽
      </span>
    </div>
  </li>
</ul>
`;
